### PR TITLE
fix: module loading, and lots of error handling

### PR DESCRIPTION
**Pull request by platypii**
This PR contains fixes to several packages:

 - handle errors thrown in `runScript`. this can happen when imports are not found.
 - handle errors in `require`. this can happen for missing npm packages.
 - fix relative path resolution. now I can drop multi-file projects and they work! this one took a lot of testing to get right.
 - handle invalid geometries. this one I noticed when I accidentally returned a function instead of jscad objects.
 - fix auto-reload for dependent files. previously it wasn't updating, due to cache key mismatch.

Basically I made a bunch of test files with: empty file, missing exports, various invalid imports, relative imports, nested folders, and anything silly I could think of. Fixed a ton of bugs, the loader code should be pretty robust now.

### CODE REVIEW

**Review by hrgdavor — APPROVED** [review]
Changes look good. Thanks for these fixes.

Would you mind to apply main.js fixes from jscad-web also to engine-test (If not I will as soon as I spare some time).

in next step we can think of moving these useful parts from `main.js` into a lib.

**Comment by platypii** [comment]
@hrgdavor I copied fixes to engine-test. Merging...

**Comment by hrgdavor** [comment]
thanks for the fixes!
